lib/info.dart

```diff
@@ -46,8 +50,26 @@ abstract class Info {
 // TODO(sigmund): add more:
 //  - inputSize: bytes used in the Dart source program
 abstract class BasicInfo implements Info {
+  static Set<int> _ids = new Set<int>();
   final InfoKind kind;
-  final int id;
+
+  int _id;
+  int get id {
+    if (_id == null) {
+      // we hope we have all of the parents set up by now...
```

Comment:
better than hope... let's make it a reality! ;-)

How about this: instead of computing ids lazily, we compute them explicitly in a pass right before serializing.

Here this would change to:
```
int _id;
int get id {
  assert(_id != null);
  return _id;
}

computeId() {
  var hash = longName(this).hashCode;
  while(!isUnique(hash)) hash++;
  _id = hash;
}
```

Below we'd call `computeId` right before dumping them out.